src/main/java/sparksoniq/spark/iterator/flowr/ForClauseSparkIterator.java

```diff
@@ -195,4 +199,66 @@ public void close() {
         }
         return rdd;
     }
+
+
+    @Override
+    public Dataset<Row> getDataFrame(DynamicContext context) {
+        this._df = null;
+
+        if (this._child == null) {
+            // create initial RDD from expression
+            JavaRDD<Item> initialRdd = this.getNewRDDFromExpression(_expression);
+
+            // define a schema
+            String schemaString = _variableName;
+            List<StructField> fields = new ArrayList<>();
+            for (String fieldName : schemaString.split(" ")) {
```

Comment:
Can you explain this to me (split, for...)? There is only one variable here and its name is the string _variableName.

Reply:
You're absolutely right that it's an overkill here. It was the code I copied from original Spark documentation and I have been preserving it as is, for future use, when we have multiple columns (it also helps me simulate multiple columns scenerio even now while experimenting with it).